crates/accelerate/src/basis/basis_translator/compose_transforms.rs

```diff
@@ -0,0 +1,206 @@
+                let doomed_nodes = dag
+                    .op_nodes(true)
+                    .filter_map(|node| {
+                        if let Some(NodeType::Operation(op)) = dag.dag().node_weight(node) {
+                            if (gate_name.as_str(), *gate_num_qubits)
+                                == (op.op.name(), op.op.num_qubits())
+                            {
+                                Some((
+                                    node,
+                                    op.params_view()
+                                        .iter()
+                                        .map(|x| x.clone_ref(py))
+                                        .collect::<SmallVec<[Param; 3]>>(),
+                                ))
+                            } else {
+                                None
+                            }
+                        } else {
+                            None
+                        }
+                    })
+                    .collect::<Vec<_>>();
+                for (node, params) in doomed_nodes {
+                    let param_mapping: HashMap<ParameterUuid, Param> = equiv_params
+                        .iter()
+                        .map(|x| ParameterUuid::from_parameter(x.to_object(py).bind(py)))
+                        .zip(params)
+                        .map(|(uuid, param)| -> PyResult<(ParameterUuid, Param)> {
+                            Ok((uuid?, param.clone_ref(py)))
+                        })
+                        .collect::<PyResult<_>>()?;
+                    let mut replacement = equiv.clone();
+                    replacement
+                        .0
+                        .assign_parameters_from_mapping(py, param_mapping)?;
+                    let replace_dag: DAGCircuit =
+                        DAGCircuit::from_circuit_data(py, replacement.0, true)?;
+                    let op_node = dag.get_node(py, node)?;
+                    dag.substitute_node_with_dag(py, op_node.bind(py), &replace_dag, None, true)?;
+                }
+            }
+        }
+    }
+    Ok(mapped_instructions)
+}
+
+fn get_example_gates(
+    py: Python,
+    dag: &DAGCircuit,
+    example_gates: Option<Box<HashMap<(String, u32), PackedInstruction>>>,
+) -> PyResult<Box<HashMap<(String, u32), PackedInstruction>>> {
+    let mut example_gates = example_gates.unwrap_or_default();
+    for node in dag.op_nodes(true) {
+        if let Some(NodeType::Operation(op)) = dag.dag().node_weight(node) {
+            example_gates.insert((op.op.name().to_string(), op.op.num_qubits()), op.clone());
+            if op.op.control_flow() {
+                let OperationRef::Instruction(inst) = op.op.view() else {
+                    panic!("Control flow op can only be an instruction")
+                };
+                let inst_bound = inst.instruction.bind(py);
+                let blocks = inst_bound.getattr("blocks")?;
```

Comment:
Yes, I am planning on using it here. However, it does throw a wrench in some of my plans because I can't necessarily verify that `CircuitData` has any calibrations for a specific instruction due to that part of the circuit living entirely on Python.

Until then I might keep this as is, but my upcoming implementation of `BasisTranslator` does use `Operation::blocks`,  thank you for that by the way.